src/Screens/WatchLater/WatchLater.js

```diff
@@ -0,0 +1,93 @@
+import "./watchLater.css";
+import Movie from "./../../Components/Movie/Movie";
+import { useProducts } from "./../../Context/products-context";
+import { useEffect } from "react";
+import { useToken } from "./../../Context/token-context";
+
+const WatchLater = () => {
+  const { state, dispatch } = useProducts();
+
+  const { encodedToken } = useToken();
+
+  useEffect(() => {
+    localStorage.setItem("LIKED_ARRAY", JSON.stringify(state.likedArray));
+    localStorage.setItem(
+      "WATCH_LATER_ARRAY",
+      JSON.stringify(state.watchLaterArray)
+    );
+    localStorage.setItem("HISTORY_ARRAY", JSON.stringify(state.historyArray));
+    localStorage.setItem("DISLIKED_ARRAY", JSON.stringify(state.dislikedArray));
+  }, [
+    state.likedArray,
+    state.watchLaterArray,
+    state.historyArray,
+    state.dislikedArray,
+  ]);
+
+  useEffect(() => {
+    fetch("/api/user/watchlater", {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+        authorization: encodedToken,
+      },
+    })
+      .then((res) => res.json())
+      .then((data) => console.log(data));
+  }, [encodedToken]);
+
+  return (
+    <div className="WatchLater">
+      {state.watchLaterArray.length > 0 && <h1>Watch later</h1>}
+      <div className="watch-later-section">
+        {state.watchLaterArray.length === 0 && (
+          <h1 className="empty-watch-later">Nothing to Watch Later!</h1>
+        )}
+        {state.watchLaterArray.length > 0 &&
+          state.watchLaterArray.map((movie) => {
+            return (
+              <Movie
+                key={movie.id}
+                source={movie.source}
```

Comment:
you can also pass  movie object and restructure these thing later in the child component this will reduce amount of code and potential errors